pkg/sql/colflow/vectorized_flow.go

```diff
@@ -947,6 +947,7 @@ func (s *vectorizedFlowCreator) setupOutput(
 						// Start a separate recording so that GetRecording will return
 						// the recordings for only the child spans containing stats.
 						ctx, span := tracing.ChildSpanRemote(ctx, "")
+						// TODO(irfansharif): How come this span is never Finish()-ed? #58721.
```

Comment:
This should be finished right away. This whole span will actually go away at some point. It's a workaround for the old SpanStats stuff before I updated it to support multiple entries per Span.

Reply:
(by "right away" I mean after line 967)